xdsl/interactive/app.py

```diff
@@ -213,17 +220,66 @@ def watch_available_pass_list(
                     ListItem(Label(value.name), name=value.name)
                 )
 
+    def get_pass_arguments(self, selected_pass_value: type[ModulePass]) -> None:
+        """
+        This function facilitates user input of pass concatenated_arg_val by navigating
+        to the AddArguments screen, and subsequently parses the returned string upon
+        screen dismissal and appends the pass to the pass_pipeline variable.
+        """
+
+        def add_pass_with_arguments_to_pass_pipeline(concatenated_arg_val: str) -> None:
+            """
+            Called when AddArguments Screen is dismissed. This function attempts to parse
+            the returned string, and if successful, adds it to the pass_pipeline variable.
+            In case of parsing failure, the AddArguments Screen is pushed, revealing the
+            Parse Error.
+            """
+            try:
+                new_pass_with_arguments = list(
+                    parse_pipeline(
+                        f"{selected_pass_value.name}{{{concatenated_arg_val}}}"
+                    )
+                )[0]
+                self.pass_pipeline = (
+                    *self.pass_pipeline,
+                    (selected_pass_value, new_pass_with_arguments),
+                )
+
+            except PassPipelineParseError as e:
+                res = f"PassPipelineParseError: {e}"
+                screen = AddArguments(TextArea(res, id="argument_text_area"))
+                self.push_screen(screen, add_pass_with_arguments_to_pass_pipeline)
```

Comment:
could you please make this a method?

Reply:
and then pass it as a lambda to `self.push_screen`

Reply:
I'm not sure that this is possible. This is a specific thing in textual where when a screen is dismissed, it somehow automatically returns the string we want in the form of the variable (`concatenated_arg_val : str`) to the`add_pass_with_arguments_to_pass_pipeline` function that is declared in the push_screen method `self.push_screen(screen, add_pass_with_arguments_to_pass_pipeline)`. It's all connected and cannot be made as a method I think.

See 'Returning Data from Screens' Section here: 
https://textual.textualize.io/guide/screens/#returning-data-from-screens